roles/keystone/tasks/main.yml

```diff
@@ -31,9 +31,42 @@
     - etype: other
       permission: r
 
-- name: install keystone service
-  upstart_service: name=keystone user=keystone cmd=/usr/local/bin/keystone-all
-                   config_dirs=/etc/keystone
+- name: set uwsgi path (source install)
+  set_fact: uwsgi_path={{ openstack_source.virtualenv_base }}/keystone/bin/uwsgi
+  when: openstack_install_method == 'source'
+  
+- name: set uwsgi path (package install)
+  set_fact: uwsgi_path={{ openstack_package.virtualenv_base }}/keystone/bin/uwsgi
+  when: openstack_install_method == 'package'
+  
+- name: install keystone uwsgi service
+  template: src=etc/init/keystone.conf
+            dest=/etc/init/keystone.conf mode=0644
+  
+- name: Creates keystone uwsgi and httpd directories
+  file: path={{ item }} state=directory
+        owner=keystone group=keystone mode=0775
+  with_items:
+    - /var/www/keystone 
+    - /etc/keystone/uwsgi
+
+- name: install keystone uwsgi service
+  template: src=etc/keystone/uwsgi/keystone.py
```

Comment:
The template source path should match the destination path (or close to it). Instead of using a src of etc/keystone/uwsgi here, instead use something like var/www/keystone/  That's a close enough match.